server/model/spans.go

```diff
@@ -323,3 +160,46 @@ func decodeChildren(parent *Span, children []encodedSpan) ([]*Span, error) {
 	}
 	return res, nil
 }
+
+func (span *Span) setMetadataAttributes() {
+	span.Attributes["name"] = span.Name
+	span.Attributes["tracetest.span.type"] = spanType(span.Attributes)
+	span.Attributes["tracetest.span.duration"] = spanDuration(*span)
+	span.Attributes["tracetest.span.startTime"] = fmt.Sprintf("%d", span.StartTime.UnixNano())
+	span.Attributes["tracetest.span.endTime"] = fmt.Sprintf("%d", span.EndTime.UnixNano())
+}
+
+func (span *Span) setTriggerResultAttributes(run *Run) {
+	result := run.TriggerResult
+
+	switch result.Type {
+	case TriggerTypeHTTP:
+		resp := result.HTTP
+		jsonheaders, _ := json.Marshal(resp.Headers)
+		span.Attributes["tracetest.response.status"] = fmt.Sprintf("%d", resp.StatusCode)
+		span.Attributes["tracetest.response.body"] = resp.Body
+		span.Attributes["tracetest.response.headers"] = string(jsonheaders)
+	case TriggerTypeGRPC:
+		resp := result.GRPC
+		jsonheaders, _ := json.Marshal(resp.Metadata)
+		span.Attributes["tracetest.response.status"] = fmt.Sprintf("%d", resp.StatusCode)
+		span.Attributes["tracetest.response.body"] = resp.Body
+		span.Attributes["tracetest.response.headers"] = string(jsonheaders)
+	}
+}
+
+func NewTracetestRootSpan(run *Run) Span {
+	span := Span{
+		ID:         IDGen.SpanID(),
+		Name:       "Tracetest Trigger",
+		StartTime:  run.CreatedAt,
+		EndTime:    run.ServiceTriggerCompletedAt,
+		Attributes: Attributes{},
+		Children:   []*Span{},
+	}
+
+	span.setMetadataAttributes()
```

Comment:
is this working correctly? The code is making a call to a reference method, but the var is not a reference